CRM/Contribute/BAO/Contribution.php

```diff
@@ -5804,31 +5774,65 @@ public static function getLastFinancialItemIds($contributionId) {
   /**
    * Create proportional entries in civicrm_entity_financial_trxn.
    *
-   * @param array $entityParams
-   *
+   * @param int $trxnID
+   * @param float $contributionTotalAmount
+   * @param string $contributionStatus
    * @param array $lineItems
-   *
-   * @param array $ftIds
-   *
+   * @param array $financialItems
    * @param array $taxItems
    *
    */
-  public static function createProportionalFinancialEntries($entityParams, $lineItems, $ftIds, $taxItems) {
-    $eftParams = array(
-      'entity_table' => 'civicrm_financial_item',
-      'financial_trxn_id' => $entityParams['trxn_id'],
-    );
+  public static function createProportionalFinancialEntries($trxnID, $contributionTotalAmount, $contributionStatus, $lineItems, $financialItems, $taxItems) {
+    $financialTxnAmount = CRM_Core_DAO::getFieldValue('CRM_Financial_DAO_FinancialTrxn', $trxnID, 'total_amount');
+
+    if ($contributionStatus == 'Partially paid' && count($financialItems) > 1) {
+      $priceFeeTotal = 0;
+      foreach (CRM_Utils_Array::collect('amount', $financialItems) as $value) {
+        $priceFeeTotal += $value;
+      }
+    }
+
     foreach ($lineItems as $key => $value) {
-      if ($value['qty'] == 0) {
+      if (!array_key_exists($value['price_field_value_id'], $financialItems)) {
         continue;
       }
-      $eftParams['entity_id'] = $ftIds[$value['price_field_value_id']];
-      $entityParams['line_item_amount'] = $value['line_total'];
-      self::createProportionalEntry($entityParams, $eftParams);
-      if (array_key_exists($value['price_field_value_id'], $taxItems)) {
-        $entityParams['line_item_amount'] = $taxItems[$value['price_field_value_id']]['amount'];
+
+      $eftParams = array(
+        'entity_table' => 'civicrm_financial_item',
+        'financial_trxn_id' => $trxnID,
+        'entity_id' => $financialItems[$value['price_field_value_id']]['financial_item_id'],
+      );
+
+      // In case of partial payment, calculate proportional amount for each line-item and created respective EFT records
+      if ($contributionStatus == 'Partially paid') {
```

Comment:
I think this will break if it will be a full payment.

Reply:
@pradpnayak can you please provide a use-case in steps?

Reply:
I didn't try but  i guess it may fail for usecase
1. Create pending contribution $100
2. Record $90 payment
2. Record $10 payment --- This will set contribution status to completed and might not record an additional entry in entity financial trxn table pointing to financial item for $10

Reply:
I was able to replicate it, thank you :) Will fix it

Reply:
@pradpnayak I have fixed it and updated the PR, also the corrected the modified unit tests. Please have a look.

Reply:
@pradpnayak we'd like to have this merged. Any remaining concerns?